pkg/core/state/notification_event.go

```diff
@@ -23,12 +24,13 @@ type NotificationEvent struct {
 // AppExecResult represent the result of the script execution, gathering together
 // all resulting notifications, state, stack and other metadata.
 type AppExecResult struct {
-	TxHash      util.Uint256
-	Trigger     trigger.Type
-	VMState     vm.State
-	GasConsumed int64
-	Stack       []stackitem.Item
-	Events      []NotificationEvent
+	TxHash         util.Uint256
+	Trigger        trigger.Type
+	VMState        vm.State
+	GasConsumed    int64
+	Stack          []stackitem.Item
+	Events         []NotificationEvent
+	FaultException error
```

Comment:
I think, string is ok here, because we use it as string anyway and `encode/decode` conversion is lossy.